src/io/sampling.jl

```diff
@@ -117,9 +161,37 @@ function sample(dom::Domain, n::Int, sampler=SobolSample(); supported_dists=Dict
 
     # Adjust samples for discrete values using flooring trick
     # Ensure unguided scenarios do not have superfluous parameter values
-    return adjust_samples(dom, spec, df)
+    return df
 end
 
+"""
+    sample_site_selection(d::Domain, n::Int)::DataFrame
+
+Create samples of only site selection parameters and rescale to distribution defined in the model spec.
+
+# Arguments
+- d : Domain.
+- n : number of samples to generate.
+
+"""
+function sample_site_selection(d::Domain, n::Int, sampler=SobolSample())::DataFrame
```

Comment:
I've changed this so it now constructs a single paramter spec in the same order as the model spec to sample from...by alternate distributions, do you mean the supported_dists variable in sample() ?

Reply:
In a way yes. There's a step to transform the samples to the distributions specified in the model spec that is missing.

Reply:
See from L94 to L99 in the below

Triangular distributions are handled as a special case because a three-element entry, $a, b, c$ is expected where these relate to the lower and upper bounds and the location of the peak in percentile terms.

https://github.com/open-AIMS/ADRIA.jl/blob/7b50c835411618f8698493cb8cd62f6a00be20e1/src/io/sampling.jl#L94

Reply:
I can see this section of code still on L136 to L142 
https://github.com/open-AIMS/ADRIA.jl/blob/6b08bde0b652ef58073fec640211d83c0d4f1e0a/src/io/sampling.jl#L136

Reply:
Okay, I see the flow now. Thank you.

Reply:
which parts are not resolved sorry?

Reply:
> The approach of combining separate samples breaks the Sobol' scheme (Sobol' sampling is not pseudo-random sampling).

> Collate all parameters for the desired combination of components first, then sample.

> I would extend the sample() function to accept one or more components for this.

```julia
function sample(d, n, components...; sampler=SobolSample())
    # Extract component parameters and process in one go in here
end
```

Reply:
sample_site_selection no longer combines sample though, it first collects the components into a spec dataframe and then samples everything at once. Do you mean just the more than one components bit for sample? It currently does one component but I can add more

Reply:
To be more specific, instead of this, where individual components are being extracted one by one then concatenated together:

```julia
crit_spec = component_params(d.model, Criteria)
env_spec = component_params(d.model, EnvironmentalLayer)

int_spec = component_params(d.model, Intervention)

# ...

sample_df = vcat(vcat(env_spec, guided_spec), crit_spec)
```

Make it so you can do all that in one go (and avoid that awkward `vcat(vcat())`

```julia
sub_spec = component_params(d.model, EnvironmentalLayer, Intervention, Criteria)
site_selection_sample = sample(sub_spec, n, sampler)
# ...
```

Reply:
Past Takuya already thought of this use case, so please make use of this method:

https://github.com/open-AIMS/ADRIA.jl/blob/043f23e825cb96a2a5ae4a92bc6573a2c0862aa0/src/Domain.jl#L390